### PR TITLE
Send and receive sms

### PR DESCRIPTION
**Pull request by abadaracco**
The application is now able to send sms, to a given number and a text provided by the user, and also to capture incoming sms and show them in the screen.

### CODE REVIEW

File: app/src/main/java/uy/com/bix/app/smsproject/classes/SmsReceiver.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	@Override
+	public void onReceive(Context context, Intent intent) {
+		// Get the data (SMS data) bound to intent
+		Bundle bundle = intent.getExtras();
+
+		SmsMessage[] msgs = null;
+
+		String str = "";
```
Comment (alebarreiro) [inline app/src/main/java/uy/com/bix/app/smsproject/classes/SmsReceiver.java]: I'm sure that you can find a better name for this variable :)

Comment (abadaracco) [inline app/src/main/java/uy/com/bix/app/smsproject/classes/SmsReceiver.java]: @alebarreiro great comment, thanks! That's what happens when you steal some code from the web

File: app/src/main/java/uy/com/bix/app/smsproject/classes/SmsReceiver.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import android.widget.Toast;
+
+public class SmsReceiver extends BroadcastReceiver {
+	public SmsReceiver() {
+	}
+
+	@Override
+	public void onReceive(Context context, Intent intent) {
+		// Get the data (SMS data) bound to intent
+		Bundle bundle = intent.getExtras();
```
Comment (reviewsbix) [inline app/src/main/java/uy/com/bix/app/smsproject/classes/SmsReceiver.java:18]: I think would be good to set 2 spaces as the default indentation.

File: app/src/main/java/uy/com/bix/app/smsproject/activity/MainActivity.java [hunk 3]
```diff
@@ -28,22 +34,79 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-        mButtonSend.setOnClickListener(new View.OnClickListener() {
+      mButtonSend.setOnClickListener(new View.OnClickListener() {
+        @Override
+        public void onClick(View v) {
+          //We get the number and the message that were inserted in the view
+          String phoneNro = mEditTextTelephone.getText().toString();
+          String smsText = mEditTextMessage.getText().toString();
+
+          String sent = "SMS_SENT";
```
Comment (reviewsbix) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/MainActivity.java]: Would be good to configure this as constants in a separated file.

File: app/src/main/java/uy/com/bix/app/smsproject/activity/MainActivity.java [hunk 4]
```diff
@@ -28,22 +34,79 @@ protected void onCreate(Bundle savedInstanceState) {
...
+          }, new IntentFilter(sent));
+
+          //---when the SMS has been delivered---
+          registerReceiver(new BroadcastReceiver(){
+            @Override
+            public void onReceive(Context arg0, Intent arg1) {
+              switch (getResultCode())
+              {
+                case Activity.RESULT_OK:
+                  Toast.makeText(getBaseContext(), "SMS delivered",
```
Comment (reviewsbix) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/MainActivity.java]: Same here, I think would be good to place "SMS delivered" inside a constants file.

Comment (abadaracco) [inline app/src/main/java/uy/com/bix/app/smsproject/activity/MainActivity.java]: I agree, I will merge this and make a new task for it

**Pull request merged**